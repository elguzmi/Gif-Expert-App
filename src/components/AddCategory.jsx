import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ( { onNewCategory } ) => {
  
  const [ inputValue , setInputValue  ]  =  useState('')  
  const onInputChange = ( ev )=>{
    //console.log(ev.target.value);
    setInputValue(ev.target.value)
  }

  const onSubmitForm = ( ev )=>{
    ev.preventDefault();
    if(inputValue.trim().length <= 1)return 
    console.log({inputValue});
    onNewCategory(inputValue.trim());
    //onAddCategories( cat => [ inputValue , ...cat  ] )
    setInputValue('')
  }
  return (
      <form action="" onSubmit={ (event)=> onSubmitForm(event) }>
        <input type="text" 
          placeholder="Buscar Gifs"
          value={inputValue}
          onChange={ onInputChange }
        />
        <button type="submit" >Agregar</button>
      </form>
    )
  }

AddCategory.propTypes ={
  onNewCategory : PropTypes.func.isRequired
}