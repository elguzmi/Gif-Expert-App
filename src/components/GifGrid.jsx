const GifGrid = ({ category }) => {
  
    return (
    <>
        <h3>{ category }</h3>
        <p>Hola mundos</p>  
    </>
  )
}

export default GifGrid
