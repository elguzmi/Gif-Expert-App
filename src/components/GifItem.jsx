import PropTypes from 'prop-types'

export const GifItem = ({title, url}) => {
  return (
    <div className='card'>
      <img src={url} alt={title} width={100} />
      <p>{title}</p>
    </div>
  )
}

// required es para que los parametros o props sean requeridos
GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
