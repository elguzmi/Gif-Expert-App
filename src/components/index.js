// Archivo de barril para exportar todos los componentes

// exportar todo lo que exporta cada componente
export * from './AddCategory'
export * from './GifItem'
export * from './GifGrid'

