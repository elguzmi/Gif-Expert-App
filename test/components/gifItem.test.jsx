import { render , screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem"; // importo el compoente

describe("Pruebas en <GifItem />", () => {

  const title = "GifItem";
  const url = "https://firebasestorage.googleapis.com/v0/b/portafolio-develop.appspot.com/o/pictures%2Fcharp.jpg?alt=media&token=1791869b-9e0c-4967-9345-e9662e0c1a3f";

  test("debe de hacer match con el snapshot", () => {      
    // Renderizar el git item
    const { container } = render(<GifItem title={title} url={url} />);

    // expect es una funcion que recibe un valor y retorna un booleano
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la imagen con el url y el alt indicado", () => {
    render(<GifItem title={title} url={url} />); // renderizar el componente
    // screen.debug();
    //console.log(screen.getByRole("img")); // consigue la imagen del screen
    //expect(screen.getByRole("img").src).toBe(url);

    const { src , alt } = screen.getByRole("img");
    expect(src).toBe(url)    
    expect(alt).toBe(alt)    
  })

  test("debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  
  }) 
  
});