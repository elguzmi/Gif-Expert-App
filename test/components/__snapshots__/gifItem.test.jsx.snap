// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="GifItem"
      src="https://firebasestorage.googleapis.com/v0/b/portafolio-develop.appspot.com/o/pictures%2Fcharp.jpg?alt=media&token=1791869b-9e0c-4967-9345-e9662e0c1a3f"
      width="100"
    />
    <p>
      GifItem
    </p>
  </div>
</div>
`;
